numba/tests/test_gil.py

```diff
@@ -164,6 +164,34 @@ def caller(a, i):
             compiled_f(a, i)
         self.check_gil_held(caller)
 
+    def test_cc_export_nogil(self):
+        from numba.pycc import CC
+        cc = CC('pycc_test_nogil')
+        @cc.export("f_cc_export", f_sig, nogil=True)
+        def f_nogil(a, indices):
+            # same as f but for cc.export
+            for idx in indices:
+                # Let another thread run
+                sleep(10 * sleep_factor)
+                a[idx] = PyThread_get_thread_ident()
+        cc.compile()
+        from pycc_test_nogil import f_cc_export
```

Comment:
I think this module isn't detected while importing on CI. Though, locally on my system it works fine. @gmarkall Any ideas how to make this work? I am trying to figure out the pattern in `compile_with_pycc.py` and `test_pycc.py` but unable to succeed. I would appreciate any hints/ideas for this. Thanks.

Reply:
I suggest moving these tests to `test_pycc` as they are testing `CC` capabilities. As a benefit of doing that, the `test_pycc`  module contains facilities/infrastructure to help with testing AOT compilation (e.g. patching `sys.modules` appropriately) which might help?

Reply:
If the `cc.export` decorated functions in these tests are moved to `numba/tests/compile_with_pycc.py` and then the checking part of unit test added to new tests in `test_pycc::TestCC` then I think it would likely "just work" as the infrastructure is already set up in there. I think addition of a `nogil` flag is similar to having a `use_nrt` setting cf. https://github.com/numba/numba/blob/974a5ca78182d2b35f32df6b395d38ef636f26aa/numba/tests/compile_with_pycc.py#L16-L17